internal/contentmanager/contentmanager.go

```diff
@@ -0,0 +1,203 @@
+package contentmanager
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/operator-framework/operator-controller/api/v1alpha1"
+	oclabels "github.com/operator-framework/operator-controller/internal/labels"
+)
+
+type ContentManager interface {
+	// ManageContent will:
+	// 1. Create a new controller-runtime cache.Cache belonging to the provided ClusterExtension
+	// 2. For each object provided:
+	//   A. Use the provided controller.Controller to establish a watch for the resource
```

Comment:
This description seems implementation-specific, not interface-generic. Can this be re-written to describe the general purpose of the function and details about contracts that implementations are expected to uphold?

Reply:
Updated the names of the functions and description. Let me know what you think!